subsys/bluetooth/mesh/shell/rpr.c

```diff
@@ -38,7 +38,7 @@ static void rpr_scan_report(struct bt_mesh_rpr_cli *cli,
 		uint8_t len, type;
 		uint8_t data[31];
 
-		len = net_buf_simple_pull_u8(adv_data) - 1;
+		len = MIN(net_buf_simple_pull_u8(adv_data) - 1, sizeof(data));
```

Comment:
```suggestion
		len = MIN(net_buf_simple_pull_u8(adv_data), sizeof(data)) - 1);
```

`len` must be at least 1 smaller than `sizeof(data)` otherwise `data[len] = '\0';` will be an invalid index (e.g. if trying to access `data[31]`

Reply:
yep, missed that. Thanks